public/app/features/admin/AdminEditUser.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { StoreState, OrgUser } from 'app/types';
+import { loadUser, clearUser } from 'app/core/components/UserEdit/state/actions';
+import { getRouteParamsId } from 'app/core/selectors/location';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { NavModel } from '@grafana/data';
+import Page from 'app/core/components/Page/Page';
+import UserProfile from 'app/core/components/UserEdit/UserProfile';
+import UserSessions from 'app/core/components/UserEdit/UserSessions';
+import NewUserPassword from 'app/core/components/UserEdit/Admin/NewUserPassword';
+import UserPermissions from 'app/core/components/UserEdit/Admin/UserPermissions';
+import UserOrganizations from 'app/core/components/UserEdit/Admin/UserOrganizations';
+
+export interface Props {
+  navModel: NavModel;
+  userId: number;
+  user: OrgUser;
+  loadUser: typeof loadUser;
+  clearUser: typeof clearUser;
+}
+
+export class AdminEditUser extends PureComponent<Props> {
+  async componentDidMount() {
+    const { userId } = this.props;
+    await this.props.loadUser(userId);
+  }
+
+  componentWillUnmount() {
+    this.props.clearUser();
+  }
+
+  render() {
+    const { navModel, userId, user } = this.props;
+    const isLoading = Object.keys(user).length === 0;
+    const adminMode = true;
+    return (
+      <Page navModel={navModel}>
+        <Page.Contents isLoading={isLoading}>
+          {!isLoading && <UserProfile adminMode={adminMode} userId={userId} />}
+          {userId && (
+            <>
+              <NewUserPassword userId={userId} />
+              {!isLoading && <UserPermissions userId={userId} />}
+              <UserOrganizations userId={userId} />
+              <UserSessions adminMode={adminMode} userId={userId} />
+            </>
+          )}
+        </Page.Contents>
+      </Page>
+    );
+  }
+}
+
+function mapStateToProps(state: StoreState) {
+  return {
+    navModel: getNavModel(state.navIndex, `global-users`),
+    userId: getRouteParamsId(state.location),
+    user: state.user.profile,
+  };
+}
+
+const mapDispatchToProps = {
+  loadUser,
```

Comment:
I don't think this transient state needs to be kept in Redux. I think we should treat Redux as a source of truth for data that is persistent during single user session. In this case loding user via Redux introduces unnecessary complexity so I suggest implementing this logic either in this component or introducing a component that's only responsibility is to provide other components with user profile.

Reply:
I didn't use it originally. I moved it to Redux because so that the Permissions component could access the profile as well without loading it twice. Please advise.

Reply:
I see. But UserPermissions is used in AdminEditUser only, which loads the user, right? So it wouldn't be loaded twice. If you load the user and keep it in a local state, then you will simply pass it via prop to UserPermissions :)